internal/controller/controllers/infraenv_controller.go

```diff
@@ -573,8 +578,18 @@ func (r *InfraEnvReconciler) updateInfraEnvStatus(
 		infraEnv.Status.ISODownloadURL = internalInfraEnv.DownloadURL
 		imageCreatedAt := metav1.NewTime(time.Time(internalInfraEnv.GeneratedAt))
 		infraEnv.Status.CreatedTime = &imageCreatedAt
+	}
 
-		// set initrd and script endpoint here so we're not changing the auth token constantly
+	existingInitrdURL, err := url.Parse(infraEnv.Status.BootArtifacts.InitrdURL)
+	if err != nil {
+		return r.handleEnsureISOErrors(ctx, log, infraEnv, err, internalInfraEnv)
+	}
+	desiredInitrdURL, err := url.Parse(infraEnv.Status.BootArtifacts.KernelURL)
```

Comment:
Why is the kernel URL the reference? Does it get updated to the correct scheme through some other process?